### ISSUE TITLE
b-card footer and header template slots not working

### ISSUE THREAD

**Issue by ChrisCalculus** [bootstrap-vue/bootstrap-vue#1680]
First of, thanks for all your hard work and time.

The b-card footer (and header) template slot appears to have stopped working in v2.0.0-rc.2.

```
<b-card>
  <strong slot="header">Some header content</strong>
  <b-row>Some content</b-row>
  <em slot="footer">Some footer content</em>
</b-card>
```

The header and footer content are not displayed (where they previously did).

- Operating system and version: Mac High Sierra
- Browser and version: Chrome Version 65.0.3325.162 (Official Build) (64-bit)
- Version of Bootstrap-Vue that you are using: v2.0.0-rc.2
- Reduced test cases: [https://jsfiddle.net/h1h9g4h2/](https://jsfiddle.net/h1h9g4h2/)

**Comment by samcgarry** [comment]
I've seen something similar on v1.5.1 but it seems related to a change between Vue JS 2.5.15 and 2.5.16. If I revert back to the older version the card slots render fine.

**Comment by ChrisCalculus** [comment]
@samcgarry interesting. In the 2.5.16 [release notes](https://github.com/vuejs/vue/releases) two bug fixes mention slots:
- fix wrongly matched named slots in functional components [62a922e](https://github.com/vuejs/vue/commit/62a922e865f5e578f67b386cb614abfc173d7851), closes [#7817](https://github.com/vuejs/vue/issues/7817)
- named slots for nested functional components [6dd73e9](https://github.com/vuejs/vue/commit/6dd73e9ee44c09f04d3f616fcce18750a55e2e4f), closes [#7710](https://github.com/vuejs/vue/issues/7710)

Hopefully this helps in debugging.

**Comment by repetier** [comment]
Strange bug. Had it with v2.0.0-rc2 and vue 2.5.2. Upgrade to vue 2.5.16 did not help, but going back to 2.5.15 seemed to do the trick for me.

**Comment by robcresswell** [comment]
"latest from npm install" - could you give us an actual version? This is completely dependent on what is specified in your package.json. I'm assuming its 2.0.0-rc2, but it'd be really helpful if you could specify the version; it's also handy for others looking at issues in the future.

**Comment by samcgarry** [comment]
I updated my bootstrap-vue to 2.0.0-rc2 and reproduced the same issue by moving from vue 2.5.15 and 2.5.16. I suspect it is probably related to the two issues in the 2.5.16 release notes that @ChrisCalculus mentioned.

**Comment by ChrisCalculus** [comment]
Hi @robcresswell, sorry yes I mentioned the version in the seond line: it's v2.0.0-rc.2 indeed.

**Comment by robcresswell** [comment]
Ah you're right, my mistake. I'll do some investigation into this later today.

**Comment by mjl** [comment]
I can confirm that it is related to vue 2.5.16; backing down to 2.5.15 makes the slots reappear with no other changes anywhere.

**Comment by lmapii** [comment]
having the same problem here, downgrading to 2.5.15 works.

**Comment by pi0** [comment]
2.5.16 problem fixed with rc.5.

**Comment by robcresswell** [comment]
@FrancescoMussi If you read the comment above your it says it was fixed with `2.0.0-rc.5`

**Comment by FrancescoMussi** [comment]
Thanks @robcresswell for fast reply!

I misunderstood that comment from @pi0 , since it was referring to Vue version I didn't get the solution was with boostrap-vue `2.0.0-rc.5`.

Thanks for clarification.
I have installed boostrap-vue `2.0.0-rc.5` and now slot header is working correctly.
